types/src/transaction/metadata.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub enum MetadataType {
+    Undefined,
+    GeneralMetadataType(GeneralMetadata),
+    TravelRuleMetadataType(TravelRuleMetadata),
+    UnstructuredStringMetadataType(UnstructuredStringMetadata),
+}
+
+// Used for versioning of general metadata
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub enum GeneralMetadata {
+    GeneralMetadataVersion0(GeneralMetadataV0),
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct GeneralMetadataV0 {
+    // Subaddress to which the funds are being sent
+    #[serde(with = "serde_bytes")]
+    to_subaddress: Option<Vec<u8>>,
+    // Subaddress from which the funds are being sent
+    #[serde(with = "serde_bytes")]
+    from_subaddress: Option<Vec<u8>>,
+    // Event sequence number of referenced payment
+    referenced_event: Option<u64>,
```

Comment:
let's add more description here for `referenced_event`, this is from the quip:
>  * referenced_event: *In the case where funds must be returned, referenced_event refers to the event sequence number of the sender’s original sent payment event.  Since refunds are just another form of p2p transfer, the referenced event field allows a refunded payment to refer back to the original payment.